autopush/senderids.py

```diff
@@ -78,17 +108,33 @@ def senderIDs(self):
             self._refresh()
         return self._senderIDs
 
+    def get(self, id):
+        """Return the associated record for a given SenderID"""
+        record = self._senderIDs.get(id)
+        if record:
+            record["senderID"] = id
+        return record
+
     def getID(self):
         """Return a randomly selected SenderID, refreshing if required"""
         self._refresh()
-        return random.choice(self._senderIDs)
+        if not len(self._senderIDs):
+            return None
+        choice = random.choice(self._senderIDs.keys())
+        record = self._senderIDs.get(choice)
+        record["senderID"] = choice
+        return record
 
     def __init__(self, args):
```

Comment:
nit: __init__ method is always the first method in a class (unless its wacky enough to have a __new__)